app/assets/javascripts/repository_columns/index.js

```diff
@@ -75,26 +89,99 @@ var RepositoryColumns = (function() {
     $('[data-attr="no-columns"]').remove();
   }
 
+  function updateListItem(column) {
+    var name = column.attributes.name;
+    $('li[data-id=' + column.id + ']').find('span').first().text(name);
+  }
+
+  function loadSpecificParams(type, params, modal) {
+    var $modal = modal;
+    var newParams = params;
+    var $statusItems;
+
+    if (type === 'RepositoryListValue') {
+      newParams.repository_column.repository_list_items_attributes = JSON.parse($('#dropdown_options').val());
+      newParams.repository_column.delimiter = $('select#delimiter').data('used-delimiter');
+    } else if (type === 'RepositoryStatusValue') {
+      $statusItems = $modal.find('.status-item-container');
+      // Load all new items
+      // Load all existing items, delete flag included
+      newParams.repository_column.repository_status_items_attributes = [];
+
+      $.each($statusItems, function(index, value) {
+        var $item = $(value);
+        var id = $item.data('id');
+        var removed = $item.data('removed');
+        var icon = $item.find('.status-item-icon').data('icon');
+        var status = $item.find('input.status-item-field').val();
+
+        if (removed && id) { // flag as item for removing
+          newParams.repository_column.repository_status_items_attributes
+            .push({ id: id, _destroy: true });
+        } else if (id) { // existing element, maybe values needs to be updated
+          newParams.repository_column.repository_status_items_attributes
+            .push({ id: id, icon: icon, status: status });
+        } else { // new element
+          newParams.repository_column.repository_status_items_attributes
+            .push({ icon: icon, status: status });
+        }
+      });
+    }
+    return newParams;
+  }
```

Comment:
I think it must be optimize, we can create inside each js file for each column same method to create params.